### PR TITLE
update several build.gradle files and dependencies, and some other stuff

### PR DESCRIPTION
**Pull request by TacoTheDank**
## General
- gradle 4.10.1 -> 5.0
- reformat changed files
- update some links and fix some english

# android build.gradle
- moved applicationId to defaultConfig because flavor applicationIds are the same
- target pie
- com.android.support 27.1.1 -> 28.0.0
- com.android.billingclient 1.1 -> 1.2
- play-services-ads 17.1.1 -> 17.1.2

# common build.gradle + dependencies.gradle
- compile with java 1.8
- com.frostwire:jlibtorrent 1.2.0.17 -> 1.2.0.18
- com.squareup.okhttp3:okhttp 3.11.0 -> 3.12.0

# desktop build.gradle
- http -> https
- com.h2database:h2 1.4.193 -> 1.4.197
- noted that org.apache.lucene:lucene-core needs to be updated to 7.x.x

Signed-off-by: Taco <SkytkRSfan3895@gmail.com>

### CODE REVIEW

File: android/build.gradle [hunk 1]
```diff
@@ -91,22 +90,18 @@ project.ext.versionCode = manifestVersionCode()
...
-
     compileSdkVersion 28
     buildToolsVersion '28.0.3'
 
     defaultConfig {
-        versionName project.ext.versionName
+        applicationId "com.frostwire.android"
         minSdkVersion 16
-        targetSdkVersion 26
+        targetSdkVersion 28
```
Comment (gubatron) [inline android/build.gradle]: sorry, can't do this yet

File: android/build.gradle [hunk 2]
```diff
@@ -123,80 +118,84 @@ android {
...
+            storeFile file("../../frostwire-tools/certs/android.keystore/frostwire.keystore")
+            storePassword ""
+            keyAlias ""
+            keyPassword ""
+        }
+    }
 }
 
 dependencies {
+    implementation 'com.android.support:animated-vector-drawable:28.0.0'
```
Comment (gubatron) [inline android/build.gradle]: I'll test your branch, but I had way too many issues when I started moving things to 28 and it was time to build a production APK

File: android/build.gradle [hunk 3]
```diff
@@ -123,80 +118,84 @@ android {
...
+    implementation 'com.android.support:customtabs:28.0.0'
+    implementation 'com.android.support:design:28.0.0'
+    implementation 'com.android.support:exifinterface:28.0.0'
+    implementation 'com.android.support:preference-v7:28.0.0'
+
+    implementation 'com.android.billingclient:billing:1.2'
```
Comment (gubatron) [inline android/build.gradle]: it requires extensive testing before making a change like this

File: android/build.gradle [hunk 4]
```diff
@@ -123,80 +118,84 @@ android {
...
-    implementation('com.mopub:mopub-sdk-banner:5.4.1@aar') { transitive = true }
-    implementation('com.mopub:mopub-sdk-interstitial:5.4.1@aar') { transitive = true }
-    implementation 'com.applovin:applovin-sdk:9.1.3'
-    implementation 'com.squareup.picasso:picasso:3.0.0-SNAPSHOT'
     implementation 'com.frostwire:jlibtorrent-android-arm:1.2.0.18'
     implementation 'com.frostwire:jlibtorrent-android-x86:1.2.0.18'
-    implementation 'com.google.android:flexbox:1.0.0'
+
+    implementation 'com.squareup.picasso:picasso:3.0.0-SNAPSHOT'
+    //implementation 'com.squareup.picasso:picasso3:3.0.0-SNAPSHOT'
```
Comment (gubatron) [inline android/build.gradle]: not sure what you did here

File: android/gradle/wrapper/gradle-wrapper.properties [hunk 5]
```diff
@@ -1,6 +1,5 @@
-#Thu Mar 17 10:42:36 EDT 2016
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-5.0-all.zip
```
Comment (gubatron) [inline android/gradle/wrapper/gradle-wrapper.properties]: I'll check this too.

Perhaps what I'll do will be to merge some of your changes.

File: common/dependencies.gradle [hunk 6]
```diff
@@ -1,5 +1,5 @@
 dependencies {
-    compile 'com.google.re2j:re2j:1.2'
-    compile "com.google.code.gson:gson:2.8.5"
-    compile 'com.squareup.okhttp3:okhttp:3.11.0'
+    implementation 'com.google.re2j:re2j:1.2'
+    implementation 'com.google.code.gson:gson:2.8.5'
+    implementation 'com.squareup.okhttp3:okhttp:3.12.0'
```
Comment (gubatron) [inline common/dependencies.gradle]: will test

File: desktop/build.gradle [hunk 7]
```diff
@@ -24,20 +24,17 @@ compileJava {
...
-    compile 'com.googlecode.gettext-commons:gettext-commons:0.9.8'
-    compile "com.h2database:h2:1.4.193"
-    compile 'org.apache.lucene:lucene-core:3.6.2'
-    compile 'net.jthink:jaudiotagger:2.2.5'
-    compile "com.frostwire:jlibtorrent:1.2.0.18"
+    implementation fileTree(dir: 'lib/jars', include: ['*.jar'])
 
-    compile fileTree(dir: 'lib/jars', include: ['*.jar'])
+    implementation 'com.googlecode.gettext-commons:gettext-commons:0.9.8'
+    implementation 'com.h2database:h2:1.4.197'
```
Comment (gubatron) [inline desktop/build.gradle]: requires testing, and perhaps even migration code

**Comment by gubatron** [comment]
684273e44 okhttp formatting and formatting merged.

(wish you had submitted these in multiple commits so I could've cherry picked)

File: common/dependencies.gradle (hunk 6 above)
Comment (gubatron) [reply common/dependencies.gradle]: just changed versions, but could not yet change the "compile" to " implementation" (and get rid of the warnings in android), as this breaks the desktop build.

```
java.lang.NoClassDefFoundError: com/google/re2j/Pattern
	at com.frostwire.regex.Pattern.buildStandardPattern(Pattern.java:537)
	at com.frostwire.regex.Pattern.<init>(Pattern.java:71)
	at com.frostwire.regex.Pattern.compile(Pattern.java:81)
	at com.frostwire.desktop.DesktopVPNMonitor.<clinit>(DesktopVPNMonitor.java:42)
	at com.frostwire.desktop.DesktopPlatform.<init>(DesktopPlatform.java:34)
	at com.limegroup.gnutella.gui.GUILoader.load(GUILoader.java:49)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at com.limegroup.gnutella.gui.Main.main(Main.java:79)
Caused by: java.lang.ClassNotFoundException: com.google.re2j.Pattern
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 11 more
```
Even building with Gradle 5.0
```
Angels-MBP:desktop gubatron$ gradle -v

------------------------------------------------------------
Gradle 5.0
------------------------------------------------------------

Build time:   2018-11-26 11:48:43 UTC
Revision:     7fc6e5abf2fc5fe0824aec8a0f5462664dbcd987

Kotlin DSL:   1.0.4
Kotlin:       1.3.10
Groovy:       2.5.4
Ant:          Apache Ant(TM) version 1.9.13 compiled on July 10 2018
JVM:          1.8.0_60 (Oracle Corporation 25.60-b23)
OS:           Mac OS X 10.14.2 x86_64

```

File: android/gradle/wrapper/gradle-wrapper.properties (hunk 5 above)
Comment (gubatron) [reply android/gradle/wrapper/gradle-wrapper.properties]: tried this with no luck, all sorts of issues with android studio.

File: android/build.gradle (hunk 1 above)
Comment (TacoTheDank) [reply android/build.gradle]: Oof, what is the reason why? Just curious :/

File: android/build.gradle (hunk 4 above)
Comment (TacoTheDank) [reply android/build.gradle]: Just a little note that picasso is going to move to picasso3 (and also where the newer snapshot releases are, so... just get ready to migrate that I guess)

File: android/build.gradle (hunk 3 above)
Comment (TacoTheDank) [reply android/build.gradle]: What about just bumping billingclient to 1.2?

File: common/dependencies.gradle (hunk 6 above)
Comment (TacoTheDank) [reply common/dependencies.gradle]: 🤔

File: desktop/build.gradle (hunk 7 above)
Comment (TacoTheDank) [reply desktop/build.gradle]: I wish more projects used semantic versioning... I figured just going from 1.4.193 to 1.4.197 wouldn't cause any breakage, but apparently it did. Sorry.

**Comment by TacoTheDank** [comment]
Apologies for not having made this easier for you...

**Comment by gubatron** [comment]
[SDK 28, android support 28, billing client 1.2 updated.](https://github.com/frostwire/frostwire/commit/7b72ea0344c815bcb4d5fbfac70fa3a567948b68)

play services update still pending.

**Pull request closed without merging**